lib/asset_cloud/base.rb

```diff
@@ -217,6 +217,8 @@ def version_details(key)
 
     def asset_class_for(key)
       klass = self.class.asset_classes[bucket_symbol_for_key(key)] || self.class.root_asset_class
+      klass = klass.call(key) if klass.is_a?(Proc)
```

Comment:
I don't think we should allow `root_asset_class` to be a `Proc`.
```ruby
klass = self.class.asset_classes[bucket_symbol_for_key(key)]
klass = klass.call(key) if klass.is_a?(Proc)
klass ||= self.class.root_asset_class
```
This also lets the proc return `nil` to fall back to the default class (might wanna add a test for this).